### ISSUE TITLE
Use Expression services from AccessCycle-configuration in FLS ServerAuthentication

### ISSUE THREAD

**Issue by HannesWell** [eclipse-passage/passage#1103]
At the moment the `org.eclipse.passage.lbc.internal.base.ServerAuthentication` has the services `BerlinProtocolExpressionParseService`, `HardwareAssessmentService`, `HardwareEnvironment`, `SimpleMapExpressionEvaluationService` hard coded.

It would be good if the mentioned class would obtain those services from the framework's access-cycle configuration. This would allow to supply custom service implementations, which is currently not possible (respectively only possible with tricks).

For the background: I want to use a custom `HardwareEnvironment` that only reads those system properties required for the current query (basically what I proposed in PR https://github.com/eclipse-passage/passage/pull/1076) that also uses OSHI 6.

I can provide a PR to enhance that.

**Comment by HannesWell** [comment]
Hello,
> Hi @HannesWell What exact version of OSHI 6 do you plan to propose?

Actually this issue/proposal was just to change the way how the mentioned services are obtained so that I can supply a custom service with another OSHI version independent of the one that Passage uses.

But if you are interested I can provide a PR to update the OSHI version of passage too.

I tested [OSHI 6.1.6](https://github.com/oshi/oshi/releases/tag/oshi-parent-6.1.6) locally which worked fine and lead to a (wall clock measured) ~30% faster license check. I also tested the latest [OSHI 6.2.0](https://github.com/oshi/oshi/releases/tag/oshi-parent-6.2.0) but unfortunately got an error on my local Windows computer, but I will report that so it hopefully gets fixed soon.

Besides performance improvements OSHI 6 also claims to be (almost) fully thread safe.

>  Does it accept Java 8?

Yes OSHI 6 also runs under Java-8.

> Will it align required JNA with Platform 2022-09?

OSHI 6.1.6 uses JNA 5.11.0, while OHI 6.2.0 uses JNA 5.12.0. According to the release notes "This release leverages improvements in JNA 5.12.0 which should significantly improve performance.". So there might be an even greater improvement, if it works...

From https://github.com/eclipse-passage/passage/issues/1084 I deduce that at least the 2022-09 train uses JNA 5.8. I don't know what's planned for 22-09 but I assume that a more recent version is welcome.

**Comment by eparovyshnaya** [comment]
Hi, @HannesWell 
Thank you for the issue. 

currently `ServerAuthentication` is _package local_ class in _internal_ package. 
It appear in AccessCycleConfiguration it must mature first. 

We have plans for it's evolution.

Have scheduled the feature to the upcoming relase.

**Comment by HannesWell** [comment]
> I also tested the latest [OSHI 6.2.0](https://github.com/oshi/oshi/releases/tag/oshi-parent-6.2.0) but unfortunately got an error on my local Windows computer, but I will report that so it hopefully gets fixed soon.

A quick update on this. In the current snapshot OSHI 6.2.1-SNAPSHOT that problem of mine is resolved.

Furthermore my mail inbox said that @eparovyshnaya has answered to this issue, but I cannot see it in the GH web-interface.
I'll try to answer that nonetheless:

Yes `ServerAuthentication` is internal and I don't call it directly. But it is called when one uses the floating license server. Therefore it is used and I noticed the 'problem'.

I did corresponding changes locally and can provide them as PR to resolve the actual intent of this issue, if you want.
